contracts/FeeBurner.sol

```diff
@@ -0,0 +1,271 @@
+     * @param _nominator nominator of initial exchange rate. See ExchangeRate struct.
+     * @param _denominator denominator of initial exchange rate. See ExchangeRate struct.
+     */
+    function addSupportFor(address _token, uint _nominator, uint _denominator)
+        public
+        onlyOperator
+        checkRate(_nominator, _denominator)
+    {
+            // Checks whether such a token hasn't been supported yet
+            require(exchangeRates[_token].blockNo == 0);
+
+            ExchangeRate memory exchangeRate = ExchangeRate(block.number, Rate(_nominator, _denominator));
+
+            previousExchangeRates[_token] = exchangeRate;
+            exchangeRates[_token] = exchangeRate;
+
+            //TODO: Should I emit an event ? Maybe sth like AddedSupport(token)
+            emit ExchangeRateChanged(_token, block.number, _nominator, _denominator);
+    }
+
+    /**
+     * @dev Sets new exchange rate for a specified token.
+     *
+     * @notice Note that the new rate will automatically take
+     *         effect after NEW_RATE_MATURITY_MARGIN number of blocks.
+     * @notice Once new rate is set it cannot be changed until it has taken effect.
+     *
+     * @param _token contract address of the ERC20 token, which rate is changed
+     * @param _nominator nominator of the new exchange rate. See ExchangeRate struct.
+     * @param _denominator denominator of the new exchange rate. See ExchangeRate struct.
+     */
+    function setExchangeRate(address _token, uint _nominator, uint _denominator)
+        public
+        onlyOperator
+        supportedToken(_token)
+        checkRate(_nominator, _denominator)
+    {
+
+        require(checkMaturityPeriodPassed(_token));
+
+        previousExchangeRates[_token] = exchangeRates[_token];
+        exchangeRates[_token] = ExchangeRate(block.number, Rate(_nominator, _denominator));
+
+        emit ExchangeRateChanged(_token, block.number, _nominator, _denominator);
+
+    }
+
+
+    /**
+     * @dev Exchanges OMGs for the given token and burns received OMGs
+     *
+     * @param _token contract address of ERC20 token, or 0 for Ether
+     * @param _nominator nominator of the rate at which the user wants to make the exchange
+     * @param _denominator denominator of the rate at which the user wants to make the exchange
+     * @param _omg_amount amount of OMGs to be exchanged and burnt
+     * @param _token_amount amount of ERC20 tokens to be exchanged
+     *
+     * @notice Rate proposed by the user must equal either to exchangeRate or to previousExchangeRate.
+     * @notice Rate cannot equal to the previousExchangeRate when the maturity period of the new rate has passed.
+     * @notice Sender may offer more OMGs than demanded by the exchange rate, the exchange/transaction is then valid.
+     */
+    function exchange(address _token, uint _nominator, uint _denominator, uint _omg_amount, uint _token_amount)
```

Comment:
I'd name the arguments _rate_nominator and _rate_denominator for a little more readability.